### PR TITLE
Remix Outlet navigator item label

### PR DESCRIPTION
**Pull request by gbalint**
**Description:**
Remix Outlet is a component with special behavior: during render it is replaced by the default exported component of the actual Remix route file.
But that route component is invisible in the navigator and does not appear in our metadata hierarchy.
As a minimal way to show which route component is rendered, I added the component name to the Outlet navigator item's label, e.g.:

image

### CODE REVIEW

File: editor/src/components/inspector/sections/layout-section/conditional-section.tsx [hunk 1]
```diff
@@ -294,7 +294,7 @@ export const ConditionalSection = React.memo(({ paths }: { paths: ElementPath[]
   )
 
   const branchLabels = useEditorState(
-    Substores.metadata,
```
Comment (bkrmendy) [inline editor/src/components/inspector/sections/layout-section/conditional-section.tsx:297]: Is this change still needed?

File: editor/src/components/canvas/remix/remix-utils.tsx [hunk 2]
```diff
@@ -383,3 +387,32 @@ export function getRoutesAndModulesFromManifest(
...
+    return null
+  }
+
+  const outletChildren = MetadataUtils.getImmediateChildrenPathsOrdered(metadata, pathTrees, path)
+  if (outletChildren.length == 0) {
+    return null
+  }
+
+  const uidsToFilePath = getAllUniqueUids(projectContents).uidsToFilePaths
+  const filePath = uidsToFilePath[EP.toUid(outletChildren[0])]
```
Comment (seanparsons) [inline editor/src/components/canvas/remix/remix-utils.tsx:407]: Minor: You could use `safeIndex` the array lookup and replace the length check above with a null check on the result.

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/navigator/navigator-item/navigator-item-wrapper.tsx [hunk 3]
```diff
@@ -97,21 +98,31 @@ const elementSupportsChildrenSelector = createCachedSelector(
 )((_, navigatorEntry) => navigatorEntryToKey(navigatorEntry))
 
 export const labelSelector = createCachedSelector(
-  (store: MetadataSubstate) => store.editor.jsxMetadata,
+  (store: MetadataSubstate & ProjectContentSubstate) => store.editor.jsxMetadata,
```
Comment (seanparsons) [inline editor/src/components/navigator/navigator-item/navigator-item-wrapper.tsx:101]: Looks like these can all be changed to the type alias created elsewhere in this PR.
```suggestion
  (store: ProjectContentAndMetadataSubstate) => store.editor.jsxMetadata,
```

**Review by seanparsons — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/inspector/sections/layout-section/conditional-section.tsx (hunk 1 above)
Comment (gbalint) [reply editor/src/components/inspector/sections/layout-section/conditional-section.tsx:297]: Yes, it is needed, because it calls the labelSelector which relies on projectContents too.
